packages/app-layout/src/AppLayoutDropdown.tsx

```diff
@@ -0,0 +1,208 @@
+import { VisuallyHidden } from '@ag.ds-next/react/a11y';
+import { Button, ButtonGroup } from '@ag.ds-next/react/button';
+import {
+	DropdownMenuDivider,
+	DropdownMenuGroup,
+	DropdownMenuItem,
+	DropdownMenuItemLink,
+	DropdownMenuItemRadio,
+	DropdownMenuPanel,
+} from '@ag.ds-next/react/dropdown-menu';
+import {
+	ArrowRightIcon,
+	AvatarIcon,
+	ChevronsLeftIcon,
+	EmailIcon,
+	ExitIcon,
+} from '@ag.ds-next/react/icon';
+import { Modal, ModalProps } from '@ag.ds-next/react/modal';
+import { NotificationBadge } from '@ag.ds-next/react/notification-badge';
+import { Text } from '@ag.ds-next/react/text';
+import { useState } from 'react';
+import { hrefs } from './utils';
+
+export interface Business {
+	partyId: number;
+	partyDisplayName: string;
+	formattedPartyExternalId: string;
+	roleDisplayName: string;
+}
+
+export interface BusinessOption extends Business {
+	selected?: boolean;
+	setSelected?: () => void;
+}
+
+export interface BusinessDetails<T extends Business> {
+	selectedBusiness?: T;
+	linkedBusinesses?: T[];
+	setSelectedBusiness: (business: T) => void;
+}
+
+const createBusinessOptions = <T extends Business>(
+	details: BusinessDetails<T>
+): BusinessOption[] => {
+	return (
+		details?.linkedBusinesses?.map((business) => ({
+			...business,
+			selected: details?.selectedBusiness?.partyId === business.partyId,
+			setSelected: () => details.setSelectedBusiness(business),
+		})) ?? []
+	);
+};
+
+export interface BusinessDropdownProps<T extends Business> {
+	businessDetails: BusinessDetails<T>;
+	unreadMessageCount?: number;
+	focusMode?: boolean;
+	onSignOutClick: () => void;
+}
+
+const ChangeBusinessModal = (props: {
+	business: BusinessOption | undefined;
+	onDismiss: ModalProps['onDismiss'];
+}) => (
+	<Modal
+		isOpen={props.business !== undefined}
+		onDismiss={props.onDismiss}
+		title="Are you sure you want to leave this page?"
```

Comment:
Do we really need this Modal?

Reply:
did you want to roll it into another release?